addons/im_livechat/static/src/bugfix/bugfix.js

```diff
@@ -7,4 +7,28 @@
 odoo.define('im_livechat/static/src/bugfix/bugfix.js', function (require) {
 'use strict';
 
+const {
+    registerInstancePatchModel,
+} = require('mail/static/src/model/model_core.js');
+
+registerInstancePatchModel('mail.chat_window', 'im_livechat/static/src/bugfix/bugfix.js', {
+
+    /**
+     * @override
+     */
+    close({ notifyServer } = {}) {
+        const thread = this.thread;
+        let args = {
+                notifyServer: notifyServer,
+        };
+        if (thread && thread['channel_type'] === 'livechat' && thread.caches[0].isLoaded && !thread['messages'].length) {
+            args = {
+                notifyServer: true,
+            };
+            thread.unpin();
+        }
+        return this._super(args);
```

Comment:
So since we are not making a generic override anyway as discussed yesterday lets remove the `return` I suggested we add earlier.
Also some cleanup in naming, but otherwise this did what it should.
- use . instead of [] to access fields
- always compare array length to an integer
- channel_type should only be asserted on channels (assert model before)

Should be something like this id I didn't make any typo:
```suggestion
    close({ notifyServer } = {}) {
        if (
            this.thread &&
            this.thread.model === 'mail.channel' &&
            this.thread.channel_type === 'livechat' &&
            this.thread.mainCache.isLoaded &&
            this.thread.messages.length === 0
        ) {
            notifyServer = true;
            this.thread.unpin();
        }
        this._super({ notifyServer });
```